### DISCUSSION TITLE
onActivated wont trigger in component

### DISCUSSION THREAD

**Discussion by perottosson** [quasarframework/quasar#14948] [General - Components / Directives / etc] [answered]
For some reason the onActivated function in vue3 wont trigger and cannot figure out why. It is not the first time I am implementing this but for some reason nothing happens. Here is my code and component

**Index.vue**

    <template>
      <MyComp />
    </template>
    
    <script setup>
    import MyComp from "../components/MyComp.vue";
    </script>

**MyComp.vue**

    <template>
      <div>{{ el }}</div>
    </template>
    
    <script setup>
    import { ref, onMounted, onActivated } from "vue";
    
    const el = ref("Hello");
    
    onMounted(() => {
      console.log("onMounted");
    });
    onActivated(() => {
      console.log("onActivated");
    });
    </script>

**Comment by pdanpdan** [comment, marked as answer]
Do you have it in a `<keep-alive>`?

**Reply by perottosson** [reply]
no, just like above

**Reply by perottosson** [reply]
onMounted runs as it should

**Reply by pdanpdan** [reply]
onActivated only works in KeepAlive

**Comment by perottosson** [comment]
it is very strange. Here are my dependencies in my package.json```
```

  "dependencies": {
    "pinia": "^2.0.11",
    "@quasar/extras": "^1.0.0",
    "core-js": "^3.6.5",
    "quasar": "^2.6.0",
    "vue": "^3.0.0",
    "vue-router": "^4.0.0"
  },
  "devDependencies": {
    "@babel/eslint-parser": "^7.13.14",
    "eslint": "^8.10.0",
    "eslint-plugin-vue": "^9.0.0",
    "eslint-webpack-plugin": "^3.1.1",
    "eslint-config-prettier": "^8.1.0",
    "prettier": "^2.5.1",
    "@quasar/app-webpack": "^3.0.0"
  },
```
